<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta name="viewport" content="width=device-width"/><meta name="next-head-count" content="2"/><link rel="preload" href="/_next/static/css/2eccd4d47c856f2b.css" as="style"/><link rel="stylesheet" href="/_next/static/css/2eccd4d47c856f2b.css" data-n-g=""/><noscript data-n-css=""></noscript><script defer="" nomodule="" src="/_next/static/chunks/polyfills-0d1b80a048d4787e.js"></script><script src="/_next/static/chunks/webpack-cb7634a8b6194820.js" defer=""></script><script src="/_next/static/chunks/framework-4556c45dd113b893.js" defer=""></script><script src="/_next/static/chunks/main-25e5079ab4bd6ecd.js" defer=""></script><script src="/_next/static/chunks/pages/_app-20edbe0b078add93.js" defer=""></script><script src="/_next/static/chunks/29107295-fbcfe2172188e46f.js" defer=""></script><script src="/_next/static/chunks/613-1e0aa2b2023820bb.js" defer=""></script><script src="/_next/static/chunks/495-bb1d5b202c02d7f2.js" defer=""></script><script src="/_next/static/chunks/471-84c36aa98dd4107c.js" defer=""></script><script src="/_next/static/chunks/81-301f760ac8107464.js" defer=""></script><script src="/_next/static/chunks/pages/questions/%5Bslug%5D-76d2c3e2d98bb08b.js" defer=""></script><script src="/_next/static/6_hwtyf_oFlFS5l_S4CTe/_buildManifest.js" defer=""></script><script src="/_next/static/6_hwtyf_oFlFS5l_S4CTe/_ssgManifest.js" defer=""></script><style data-styled="" data-styled-version="5.3.5">.eZIPKL code{padding:5px;color:hsl(210deg 8% 15%);background-color:hsl(210deg 8% 90%);border-radius:3px;}/*!sc*/
data-styled.g5[id="sc-c4b0431a-0"]{content:"eZIPKL,"}/*!sc*/
</style></head><body><div id="__next"><div class="sc-9099c029-0 cIPEih"><header><nav class="bg-white border-gray-200 px-4 lg:px-6 py-2.5 dark:bg-gray-800"><div class="flex flex-wrap justify-between items-center mx-auto max-w-screen-xl"><a class="flex items-center" href="/"><img src="https://flowbite.com/docs/images/logo.svg" class="mr-3 h-6 sm:h-9" alt="Flowbite Logo"/><span class="self-center text-xl font-semibold whitespace-nowrap dark:text-white">Solution Hunter</span></a><div class="flex items-center lg:order-2"><button data-collapse-toggle="mobile-menu-2" type="button" class="inline-flex items-center p-2 ml-1 text-sm text-gray-500 rounded-lg lg:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600" aria-controls="mobile-menu-2" aria-expanded="false"><span class="sr-only">Open main menu</span><svg class="w-6 h-6" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z" clip-rule="evenodd"></path></svg><svg class="hidden w-6 h-6" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z" clip-rule="evenodd"></path></svg></button></div><div class="hidden justify-between items-center w-full lg:flex lg:w-auto lg:order-1" id="mobile-menu-2"><ul class="flex flex-col mt-4 font-medium lg:flex-row lg:space-x-8 lg:mt-0"><li><a class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-blue-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700" aria-current="page" href="/">Home</a></li><li><a class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-blue-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700" href="/questions?tab=news">Questions</a></li><li><a class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-blue-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700" href="/post?tab=news">Post</a></li><li><a class="block py-2 pr-4 pl-3 text-gray-700 border-b border-gray-100 hover:bg-gray-50 lg:hover:bg-transparent lg:border-0 lg:hover:text-blue-700 lg:p-0 dark:text-gray-400 lg:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white lg:dark:hover:bg-transparent dark:border-gray-700" href="/questions/importing-installed-package-from-script-raises-%22attributeerror%3A-module-has-no-attribute%22-or-%22importerror%3A-cannot-import-name%22-1657387790607#">Coding</a></li></ul></div></div></nav></header><div class="main-content"><div class="sc-c5440139-0 figLul question my-5"><div class="sc-c4b0431a-0 eZIPKL flex question-header items-center justify-center"><div class="rounded-xl border p-5 shadow-md w-9/12 bg-white"><div class="flex w-full items-center justify-between border-b pb-3"><div class="flex items-center space-x-3"><div class="text-lg font-bold text-slate-700"><a href="/questions/importing-installed-package-from-script-raises-%22attributeerror:-module-has-no-attribute%22-or-%22importerror:-cannot-import-name%22-1657387790607">Importing installed package from script raises &quot;AttributeError: module has no attribute&quot; or &quot;ImportError: cannot import name&quot;</a></div></div><div class="flex flex-wrap h-auto justify-end items-center space-x-8"><a class="rounded-2xl border bg-neutral-100 px-3 py-1 text-sm font-semibold" href="/questions/tag/shadowing">shadowing</a></div></div><div class="question-content mt-5">
                
<p>I have a script named <code>requests.py</code> that imports the requests package.  The script either can't access attributes from the package, or can't import them.  Why isn't this working and how do I fix it?</p>

<p>The following code raises an <code>AttributeError</code>.</p>

<pre class="lang-py s-code-block"><code class="hljs language-python"><span class="hljs-keyword">import</span> requests

res = requests.get(<span class="hljs-string">'http://www.google.ca'</span>)
<span class="hljs-built_in">print</span>(res)
</code></pre>

<pre class="lang-none s-code-block"><code>Traceback (most recent call last):
  File "/Users/me/dev/rough/requests.py", line 1, in &lt;module&gt;
    import requests
  File "/Users/me/dev/rough/requests.py", line 3, in &lt;module&gt;
    requests.get('http://www.google.ca')
AttributeError: module 'requests' has no attribute 'get'
</code></pre>

<p>The following code raises an <code>ImportError</code>.</p>

<pre class="lang-py s-code-block"><code class="hljs language-python"><span class="hljs-keyword">from</span> requests <span class="hljs-keyword">import</span> get

res = get(<span class="hljs-string">'http://www.google.ca'</span>)
<span class="hljs-built_in">print</span>(res)
</code></pre>

<pre class="lang-none s-code-block"><code>Traceback (most recent call last):
  File "requests.py", line 1, in &lt;module&gt;
    from requests import get
  File "/Users/me/dev/rough/requests.py", line 1, in &lt;module&gt;
    from requests import get
ImportError: cannot import name 'get'
</code></pre>

<p>Or code that imports from a module inside the <code>requests</code> package:</p>

<pre class="lang-py s-code-block"><code class="hljs language-python"><span class="hljs-keyword">from</span> requests.auth <span class="hljs-keyword">import</span> AuthBase
</code></pre>

<pre class="lang-none s-code-block"><code>Traceback (most recent call last):
  File "requests.py", line 1, in &lt;module&gt;
    from requests.auth import AuthBase
  File "/Users/me/dev/rough/requests.py", line 1, in &lt;module&gt;
    from requests.auth import AuthBase
ImportError: No module named 'requests.auth'; 'requests' is not a package
</code></pre>
    </div></div></div><div class="sc-c4b0431a-2 cRqwQe"><div class="flex mt-5 answer items-center justify-center py-5"><div class="rounded-xl border p-10 shadow-md w-9/12 bg-white"><h4 class="text-4xl font-semibold mb-5">Solution 1</h4><div class=" items-center justify-between"><div class=" space-x-4 md:space-x-8">
<p>This happens because your local module named <code>requests.py</code> shadows the installed <code>requests</code> module you are trying to use.  The current directory is prepended to <code>sys.path</code>, so the local name takes precedence over the installed name.</p>

<p>An extra debugging tip when this comes up is to look at the Traceback carefully, and realize that the name of your script in question is matching the module you are trying to import: </p>

<p>Notice the name you used in your script:</p>

<pre class="lang-py s-code-block"><code class="hljs language-python">File <span class="hljs-string">"/Users/me/dev/rough/requests.py"</span>, line <span class="hljs-number">1</span>, <span class="hljs-keyword">in</span> &lt;module&gt;
</code></pre>

<p>The module you are trying to import: <code>requests</code></p>

<p>Rename your module to something else to avoid the name collision.</p>

<p>Python may generate a <code>requests.pyc</code> file next to your <code>requests.py</code> file (in the <code>__pycache__</code> directory in Python 3).  Remove that as well after your rename, as the interpreter will still reference that file, re-producing the error. However, the <code>pyc</code> file in <code>__pycache__</code> <em>should</em> not affect your code if the <code>py</code> file has been removed.</p>

<p>In the example, renaming the file to <code>my_requests.py</code>, removing <code>requests.pyc</code>, and running again successfully prints <code>&lt;Response [200]&gt;</code>.</p>
    </div></div></div></div><div class="flex mt-5 answer items-center justify-center py-5"><div class="rounded-xl border p-10 shadow-md w-9/12 bg-white"><h4 class="text-4xl font-semibold mb-5">Solution 2</h4><div class=" items-center justify-between"><div class=" space-x-4 md:space-x-8">
<p>For the writer of the original question, and for those people searching on the AttributeError: module has no attribute string, then the common explanation as per the accepted answer, is that a user-created script has a name-clash with a library filename. Note, however, that the trouble might not be in the name of the script that generates the error (as it was in the above case), nor in any of the names of the library modules explicitly imported by that script. It might take a little detective work to figure out which file is causing the problem. </p>

<p>As an example to illustrate the problem, imagine that you are creating a script that uses the "decimal" library for accurate floating-point calculations with decimal numbers, and you name your script "<code>mydecimal.py</code>" that contains the line "<code>import decimal</code>". There's no problem with any of that but you find that it raises this error:</p>

<pre class="lang-py s-code-block"><code class="hljs language-python">AttributeError: <span class="hljs-string">'module'</span> <span class="hljs-built_in">object</span> has no attribute <span class="hljs-string">'Number'</span>
</code></pre>

<p>This would happen if you had <em>previously</em> written a script called "<code>numbers.py</code>" because the "decimal" library calls on the standard library "numbers" but finds your old script instead. Even if you had deleted that, it might not end the problem because python might have converted that into bytecode and stored it in a cache as "<code>numbers.pyc</code>", so you'd have to hunt that down as well.</p>
    </div></div></div></div><div class="flex mt-5 answer items-center justify-center py-5"><div class="rounded-xl border p-10 shadow-md w-9/12 bg-white"><h4 class="text-4xl font-semibold mb-5">Solution 3</h4><div class=" items-center justify-between"><div class=" space-x-4 md:space-x-8">
<p>Python is looking for the requests object within your <code>requests.py</code> module.</p>
<p>Either <strong>RENAME</strong> that file to something else or use</p>
<pre class="lang-py s-code-block"><code class="hljs language-python"><span class="hljs-keyword">from</span> __future__ <span class="hljs-keyword">import</span> absolute_import 
</code></pre>
<p>at the top of your <code>requests.py</code> module.</p>
    </div></div></div></div><div class="flex mt-5 answer items-center justify-center py-5"><div class="rounded-xl border p-10 shadow-md w-9/12 bg-white"><h4 class="text-4xl font-semibold mb-5">Solution 4</h4><div class=" items-center justify-between"><div class=" space-x-4 md:space-x-8">
<p>By <code>import requests</code> you import the file requests.py, not the module.
Simply rename your file.</p>
    </div></div></div></div></div></div></div></div></div><script id="__NEXT_DATA__" type="application/json">{"props":{"pageProps":{"data":{"answer":["\n\u0026lt;p\u0026gt;This happens because your local module named \u0026lt;code\u0026gt;requests.py\u0026lt;/code\u0026gt; shadows the installed \u0026lt;code\u0026gt;requests\u0026lt;/code\u0026gt; module you are trying to use.  The current directory is prepended to \u0026lt;code\u0026gt;sys.path\u0026lt;/code\u0026gt;, so the local name takes precedence over the installed name.\u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;An extra debugging tip when this comes up is to look at the Traceback carefully, and realize that the name of your script in question is matching the module you are trying to import: \u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;Notice the name you used in your script:\u0026lt;/p\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;File \u0026lt;span class=\u0026quot;hljs-string\u0026quot;\u0026gt;\u0026quot;/Users/me/dev/rough/requests.py\u0026quot;\u0026lt;/span\u0026gt;, line \u0026lt;span class=\u0026quot;hljs-number\u0026quot;\u0026gt;1\u0026lt;/span\u0026gt;, \u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;in\u0026lt;/span\u0026gt; \u0026amp;lt;module\u0026amp;gt;\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;p\u0026gt;The module you are trying to import: \u0026lt;code\u0026gt;requests\u0026lt;/code\u0026gt;\u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;Rename your module to something else to avoid the name collision.\u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;Python may generate a \u0026lt;code\u0026gt;requests.pyc\u0026lt;/code\u0026gt; file next to your \u0026lt;code\u0026gt;requests.py\u0026lt;/code\u0026gt; file (in the \u0026lt;code\u0026gt;__pycache__\u0026lt;/code\u0026gt; directory in Python 3).  Remove that as well after your rename, as the interpreter will still reference that file, re-producing the error. However, the \u0026lt;code\u0026gt;pyc\u0026lt;/code\u0026gt; file in \u0026lt;code\u0026gt;__pycache__\u0026lt;/code\u0026gt; \u0026lt;em\u0026gt;should\u0026lt;/em\u0026gt; not affect your code if the \u0026lt;code\u0026gt;py\u0026lt;/code\u0026gt; file has been removed.\u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;In the example, renaming the file to \u0026lt;code\u0026gt;my_requests.py\u0026lt;/code\u0026gt;, removing \u0026lt;code\u0026gt;requests.pyc\u0026lt;/code\u0026gt;, and running again successfully prints \u0026lt;code\u0026gt;\u0026amp;lt;Response [200]\u0026amp;gt;\u0026lt;/code\u0026gt;.\u0026lt;/p\u0026gt;\n    ","\n\u0026lt;p\u0026gt;For the writer of the original question, and for those people searching on the AttributeError: module has no attribute string, then the common explanation as per the accepted answer, is that a user-created script has a name-clash with a library filename. Note, however, that the trouble might not be in the name of the script that generates the error (as it was in the above case), nor in any of the names of the library modules explicitly imported by that script. It might take a little detective work to figure out which file is causing the problem. \u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;As an example to illustrate the problem, imagine that you are creating a script that uses the \u0026quot;decimal\u0026quot; library for accurate floating-point calculations with decimal numbers, and you name your script \u0026quot;\u0026lt;code\u0026gt;mydecimal.py\u0026lt;/code\u0026gt;\u0026quot; that contains the line \u0026quot;\u0026lt;code\u0026gt;import decimal\u0026lt;/code\u0026gt;\u0026quot;. There\u0026apos;s no problem with any of that but you find that it raises this error:\u0026lt;/p\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;AttributeError: \u0026lt;span class=\u0026quot;hljs-string\u0026quot;\u0026gt;\u0026apos;module\u0026apos;\u0026lt;/span\u0026gt; \u0026lt;span class=\u0026quot;hljs-built_in\u0026quot;\u0026gt;object\u0026lt;/span\u0026gt; has no attribute \u0026lt;span class=\u0026quot;hljs-string\u0026quot;\u0026gt;\u0026apos;Number\u0026apos;\u0026lt;/span\u0026gt;\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;p\u0026gt;This would happen if you had \u0026lt;em\u0026gt;previously\u0026lt;/em\u0026gt; written a script called \u0026quot;\u0026lt;code\u0026gt;numbers.py\u0026lt;/code\u0026gt;\u0026quot; because the \u0026quot;decimal\u0026quot; library calls on the standard library \u0026quot;numbers\u0026quot; but finds your old script instead. Even if you had deleted that, it might not end the problem because python might have converted that into bytecode and stored it in a cache as \u0026quot;\u0026lt;code\u0026gt;numbers.pyc\u0026lt;/code\u0026gt;\u0026quot;, so you\u0026apos;d have to hunt that down as well.\u0026lt;/p\u0026gt;\n    ","\n\u0026lt;p\u0026gt;Python is looking for the requests object within your \u0026lt;code\u0026gt;requests.py\u0026lt;/code\u0026gt; module.\u0026lt;/p\u0026gt;\n\u0026lt;p\u0026gt;Either \u0026lt;strong\u0026gt;RENAME\u0026lt;/strong\u0026gt; that file to something else or use\u0026lt;/p\u0026gt;\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;\u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;from\u0026lt;/span\u0026gt; __future__ \u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;import\u0026lt;/span\u0026gt; absolute_import \n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\u0026lt;p\u0026gt;at the top of your \u0026lt;code\u0026gt;requests.py\u0026lt;/code\u0026gt; module.\u0026lt;/p\u0026gt;\n    ","\n\u0026lt;p\u0026gt;By \u0026lt;code\u0026gt;import requests\u0026lt;/code\u0026gt; you import the file requests.py, not the module.\nSimply rename your file.\u0026lt;/p\u0026gt;\n    "],"id":374,"title":"Importing installed package from script raises \"AttributeError: module has no attribute\" or \"ImportError: cannot import name\"","content":"\n                \n\u0026lt;p\u0026gt;I have a script named \u0026lt;code\u0026gt;requests.py\u0026lt;/code\u0026gt; that imports the requests package.  The script either can\u0026apos;t access attributes from the package, or can\u0026apos;t import them.  Why isn\u0026apos;t this working and how do I fix it?\u0026lt;/p\u0026gt;\n\n\u0026lt;p\u0026gt;The following code raises an \u0026lt;code\u0026gt;AttributeError\u0026lt;/code\u0026gt;.\u0026lt;/p\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;\u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;import\u0026lt;/span\u0026gt; requests\n\nres = requests.get(\u0026lt;span class=\u0026quot;hljs-string\u0026quot;\u0026gt;\u0026apos;http://www.google.ca\u0026apos;\u0026lt;/span\u0026gt;)\n\u0026lt;span class=\u0026quot;hljs-built_in\u0026quot;\u0026gt;print\u0026lt;/span\u0026gt;(res)\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-none s-code-block\u0026quot;\u0026gt;\u0026lt;code\u0026gt;Traceback (most recent call last):\n  File \u0026quot;/Users/me/dev/rough/requests.py\u0026quot;, line 1, in \u0026amp;lt;module\u0026amp;gt;\n    import requests\n  File \u0026quot;/Users/me/dev/rough/requests.py\u0026quot;, line 3, in \u0026amp;lt;module\u0026amp;gt;\n    requests.get(\u0026apos;http://www.google.ca\u0026apos;)\nAttributeError: module \u0026apos;requests\u0026apos; has no attribute \u0026apos;get\u0026apos;\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;p\u0026gt;The following code raises an \u0026lt;code\u0026gt;ImportError\u0026lt;/code\u0026gt;.\u0026lt;/p\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;\u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;from\u0026lt;/span\u0026gt; requests \u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;import\u0026lt;/span\u0026gt; get\n\nres = get(\u0026lt;span class=\u0026quot;hljs-string\u0026quot;\u0026gt;\u0026apos;http://www.google.ca\u0026apos;\u0026lt;/span\u0026gt;)\n\u0026lt;span class=\u0026quot;hljs-built_in\u0026quot;\u0026gt;print\u0026lt;/span\u0026gt;(res)\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-none s-code-block\u0026quot;\u0026gt;\u0026lt;code\u0026gt;Traceback (most recent call last):\n  File \u0026quot;requests.py\u0026quot;, line 1, in \u0026amp;lt;module\u0026amp;gt;\n    from requests import get\n  File \u0026quot;/Users/me/dev/rough/requests.py\u0026quot;, line 1, in \u0026amp;lt;module\u0026amp;gt;\n    from requests import get\nImportError: cannot import name \u0026apos;get\u0026apos;\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;p\u0026gt;Or code that imports from a module inside the \u0026lt;code\u0026gt;requests\u0026lt;/code\u0026gt; package:\u0026lt;/p\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-py s-code-block\u0026quot;\u0026gt;\u0026lt;code class=\u0026quot;hljs language-python\u0026quot;\u0026gt;\u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;from\u0026lt;/span\u0026gt; requests.auth \u0026lt;span class=\u0026quot;hljs-keyword\u0026quot;\u0026gt;import\u0026lt;/span\u0026gt; AuthBase\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n\n\u0026lt;pre class=\u0026quot;lang-none s-code-block\u0026quot;\u0026gt;\u0026lt;code\u0026gt;Traceback (most recent call last):\n  File \u0026quot;requests.py\u0026quot;, line 1, in \u0026amp;lt;module\u0026amp;gt;\n    from requests.auth import AuthBase\n  File \u0026quot;/Users/me/dev/rough/requests.py\u0026quot;, line 1, in \u0026amp;lt;module\u0026amp;gt;\n    from requests.auth import AuthBase\nImportError: No module named \u0026apos;requests.auth\u0026apos;; \u0026apos;requests\u0026apos; is not a package\n\u0026lt;/code\u0026gt;\u0026lt;/pre\u0026gt;\n    ","slug":"importing-installed-package-from-script-raises-\"attributeerror:-module-has-no-attribute\"-or-\"importerror:-cannot-import-name\"-1657387790607","postType":"QUESTION","createdAt":"2022-07-09T17:29:50.000Z","updatedAt":"2022-07-09T17:29:50.000Z","tags":[{"id":2055,"name":"shadowing","slug":"shadowing","createdAt":"2022-07-09T17:29:50.000Z","updatedAt":"2022-07-09T17:29:50.000Z","Questions_Tags":{"questionId":374,"tagId":2055}}]}},"__N_SSG":true},"page":"/questions/[slug]","query":{"slug":"importing-installed-package-from-script-raises-\"attributeerror:-module-has-no-attribute\"-or-\"importerror:-cannot-import-name\"-1657387790607"},"buildId":"6_hwtyf_oFlFS5l_S4CTe","isFallback":false,"gsp":true,"scriptLoader":[]}</script></body></html>